engine/src/main/java/io/zeebe/engine/processor/AsyncSnapshotDirector.java

```diff
@@ -78,17 +78,75 @@ protected void onActorStarting() {
     commitCondition = actor.onCondition(getConditionNameForPosition(), this::onCommitCheck);
     logStream.registerOnCommitPositionUpdatedCondition(commitCondition);
 
-    lastValidSnapshotPosition = snapshotController.getLastValidSnapshotPosition();
-    LOG.debug(
-        "The position of the last valid snapshot is '{}'. Taking snapshots beyond this position.",
-        lastValidSnapshotPosition);
+    lastValidSnapshotPosition = -1;
+    actor.runOnCompletionBlockingCurrentPhase(
+        streamProcessor.getLastProcessedPositionAsync(),
+        (position, error) -> {
+          if (error == null) {
+            lastValidSnapshotPosition = position;
+            LOG.debug(
+                "The position of the last valid snapshot is '{}'. Taking snapshots beyond this position.",
+                lastValidSnapshotPosition);
+          } else {
+            LOG.error(ERROR_MSG_ON_RESOLVE_PROCESSED_POS, error);
```

Comment:
Is it correct to continue from here?

Reply:
The worse that can happen is that we might take a snapshot when we don't need to, since we fallback to using `-1` as the last valid snapshot position. So if we had a snapshot at position 5, and we process nothing, we will have two snapshots at position 5 (but one with a higher timestamp so they can still be distinguished). It's useless work but I think it's not incorrect. We could also fail, but I'm not sure how is the recovery at the moment, so this seemed like a good compromise.

Reply:
So would could also just skip this and determine the position on the first next snapshot we take?

Reply:
How would you determine this later? Once we want to take a snapshot, how do we distinguish whether the stream processor has processed events or not?

Reply:
Maybe we dont need to?